website/static/js/registrationUtils.js

```diff
@@ -656,63 +652,64 @@ RegistrationEditor.prototype.updateData = function(response) {
-//RegistrationEditor.prototype.submitForReview = function() {
-//    var self = this;
-//
-//    var messages = self.draft().messages;
-//    bootbox.confirm(messages.beforeSubmitForApproval, function(result) {
-//	if(result) {
-//	    $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {}).then(function() {
-//		bootbox.dialog({
-//                    closeButton: false,
-//		    message: messages.afterSubmitForApproval,
-//		    title: 'Pre-Registration Prize Submission',
-//		    buttons: {
-//                        registrations: {
-//                            label: 'Return to registrations page',
-//                            className: 'btn-primary pull-right',
-//                            callback: function() {
-//                                window.location.href = self.draft().urls.registrations;
-//                            }
-//                        }
-//		    }
-//		});
-//            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
-//	}
-//    });
-//};
-// TODO: uncomment when we allow submitting for review
-//RegistrationEditor.prototype.submit = function() {
-//    var self = this;
-//
-//    var currentNode = window.contextVars.node;
-//    var currentUser = window.contextVars.currentUser;
-//
-//    var messages = self.draft().messages;
-//    bootbox.confirm(messages.beforeSubmitForApproval, function(result) {
-//        if(result) {
-//            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {
-//                node: currentNode,
-//                auth: currentUser
-//            }).then(function() {
-//                bootbox.dialog({
-//                    message: messages.afterSubmitForApproval,
-//                    title: 'Pre-Registration Prize Submission',
-//                    buttons: {
-//                        registrations: {
-//                            label: 'Return to registrations page',
-//                            className: 'btn-primary pull-right',
-//                            callback: function() {
-//                                window.location.href = self.draft().urls.registrations;
-//                            }
-//                        }
-//                    }
-//                });
-//            }).fail($osf.growl.bind(null, 'Error submitting for review', language.submitForReviewFail));
-//        }
-//    });
-//};
+RegistrationEditor.prototype.submitForReview = function() {
+    var self = this;
+
+    var draft = self.draft();
+    var metaSchema = draft.metaSchema;
+    var messages = metaSchema.messages;
+    var beforeSubmitForApprovalMessage = messages.beforeSubmitForApproval || '';
+    var afterSubmitForApprovalMessage = messages.afterSubmitForApproval || '';
+
+    bootbox.confirm(beforeSubmitForApprovalMessage, function(confirmed) {
+        if (confirmed) {
+            $osf.postJSON(self.urls.submit.replace('{draft_pk}', self.draft().pk), {}).then(function() {
```

Comment:
Typically `.then` takes a pair of handler functions. Since you're already specifying `fail` directly, perhaps change `then` to `done` here?